apps/hash-api/src/graph/util.ts

```diff
@@ -128,6 +128,10 @@ const getOrCreateOwningAccountGroupId = async (
 export const ensureAccountGroupOrgsExist = async (params: {
   context: ImpureGraphContext;
 }) => {
+  if (isSelfHostedInstance) {
+    return;
+  }
+
```

Comment:
I think it'd be cleaner to do just not call this function, which would mean moving this check to `ensureSystemGraphIsInitialized`:
```TS
export const ensureSystemGraphIsInitialized = async (params: {
  logger: Logger;
  context: ImpureGraphContext;
}) => {
  await ensureSystemAccountExists(params);

  await ensureSystemTypesExist(params);

  if (isSelfHostedInstance) {
    await ensureAccountGroupOrgsExist(params);
  }

  await ensureSystemEntitiesExists(params);
};

```

Reply:
My thinking behind putting it here was that this file contains the other logic for `isSelfHostedInstance` and it's therefore easier to see what is / isn't happening if the instance is self hosted. Ideally we would have a much cleaner process for creating system types which didn't require more than one check

Reply:
Captured as H-1233 for follow-up triage.

Reply:
Understood, I just think it'll be harder to figure out when viewing `ensureSystemGraphIsInitialized` to figure out which of the functions are actually executing. But I don't feel very strongly so happy to leave as is.